File: python/src/models/transfusion_center.py
from sqlalchemy import Column, Integer, String
from sqlalchemy.orm import relationship
from database import Base


class TransfusionCenter(Base):
    __tablename__ = "transfusion_center"
    id_center = Column(Integer, primary_key=True, nullable=False, autoincrement=True)
    name = Column(String(50), nullable=False)
    employees = relationship("Employee", back_populates="transfusion_center")
    requests = relationship("Request", back_populates="transfusion_center")

File: nodejs/database/models/transfusion_center.js
'use strict';

module.exports = (sequelize, DataTypes) => {
    const transfusion_center = sequelize.define('transfusion_center', {
        id_center: {
            type: DataTypes.INTEGER,
            primaryKey: true,
            autoIncrement: true
        },
        name: {
            type: DataTypes.STRING(50),
        }
    }, {
        freezeTableName: true,
        timestamps: false,
        underscored: true,
        name: {
            singular: 'transfusion_center',
            plural: 'transfusion_center'
        }
    });

    transfusion_center.associate = models => {
        transfusion_center.hasMany(models.request, {
            foreignKey: 'id_center',
            onDelete: 'cascade'
        });

        transfusion_center.hasMany(models.employee, {
            foreignKey: 'id_center',
            onDelete: 'cascade'
        });
    };

    return transfusion_center;
};
